src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,179 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        String query = "SELECT * FROM orders WHERE user_id = ? AND deleted = FALSE";
+        List<Order> userOrders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement statement = connection.prepareStatement(query)) {
+                statement.setLong(1, userId);
+                ResultSet resultSet = statement.executeQuery();
+                while (resultSet.next()) {
+                    userOrders.add(getOrderFromResultSet(resultSet));
+                }
+            }
+            for (Order order : userOrders) {
+                order.setProducts(getProductListByOrderId(order.getId()));
+            }
```

Comment:
There is a connection inside of another connection.
Let's move this logic down after you close the first connection.

Reply:
\+, you're closeing statement but not connection, so either close connection or pass connection into method and don't create a new one